### ISSUE TITLE
make_nth_sibling() function

### ISSUE THREAD

**Issue by wyhinton** [iwburns/id-tree#90]
I think that if a ```make_first_sibling()``` and a ```make_last_sibling()``` are available, then there should also be a ```make_nth_sibling()``` function. I'm currently creating a photoshop style layer grouping system for my rust app, using id-tree to model my tree of layers and layer groups, so I need to re-order siblings. I think this would be a pretty common use case and would make sense for the library. 

How I imagine this working:
```
   tree.make_nth_sibling(id: &NodeId, pos: u32)
 
    //        Z
    //     / / \ \
    //    A  B  C  D
  tree.make_nth_siblig(A, 2)->
    //        Z
    //     / / \ \
    //    B  C  A  D
```
Also see my stack overflow question:
https://stackoverflow.com/questions/66001937/rearrange-sibling-nodes-in-id-tree-rust?noredirect=1#comment116815312_66001937

**Comment by iwburns** [comment]
Thanks for the suggestion! This does sound like a nice feature to have.

Sadly I've been a bit preoccupied with other things lately and I haven't had much time to move this project forward. I'd be perfectly willing to accept a PR for this change if you are willing to provide one (or if someone else is interested as well of course).

I think there are probably a few minor design questions/ideas that need to be explored here; please feel free to discuss those things here if you'd like.

**Comment by wyhinton** [comment]
Added a PR based on the answer I got here:
https://stackoverflow.com/questions/66001937/rearrange-sibling-nodes-in-id-tree-rust

**Comment by iwburns** [comment]
@wyhinton thanks so much for the contribution! I'll take a look at it here shortly.

**Comment by iwburns** [comment]
resolved in #91
